proof/courier.go

```diff
@@ -632,53 +632,56 @@ func (b *BackoffHandler) initialDelay(ctx context.Context,
 	return nil
 }
 
-// Exec attempts to execute the given target function using a repeating backoff
-// time delayed strategy. The backoff strategy is used to ensure that we don't
-// spam the courier service with proof delivery attempts.
-func (b *BackoffHandler) Exec(ctx context.Context, proofLocator Locator,
-	targetFunc func() error, subscriberEvent func(fn.Event)) error {
+// Exec attempts to execute the given proof transfer function using a repeating
+// backoff time delayed strategy. The backoff strategy is used to ensure that we
+// don't spam the courier service with proof transfer attempts.
+func (b *BackoffHandler) Exec(ctx context.Context,
+	proofLocator Locator, transferType TransferType,
+	transferFunc func() ([]byte, error),
```

Comment:
Perhaps pull out transfer func and into a top level type def so this is a bit easier to read? Eg: why can it now also return a blob?

Reply:
I'll use a closure variable and keep the target function signature minimal.